### PR TITLE
Bc545 uid

### PR DESCRIPTION
**Pull request by MattTaylorDLS**
PR to look at UID creation to start the discussion on it. It's not really as elegant as i'd wanted it to be, but it works...

### CODE REVIEW

File: frameProcessor/include/FrameProcessorDefinitions.h [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace FrameProcessor
   /**
    * Enumeration to store the pixel type of the incoming image
    */
-  enum PixelType { pixel_raw_8bit, pixel_raw_16bit, pixel_float32, pixel_raw_64bit };
+  enum PixelType { pixel_raw_8bit, pixel_raw_16bit, pixel_raw_32bit, pixel_raw_64bit, pixel_float };
```
Comment (GDYendell) [inline frameProcessor/include/FrameProcessorDefinitions.h]: I don't think it makes sense for these to be called "pixel" anymore. Maybe we call the enum DataType and remove pixel_ from the front of the elements?

File: frameProcessor/include/Frame.h [hunk 2]
```diff
@@ -35,6 +36,28 @@ typedef struct
...
 
+
+/** Frame Parameter Value */
+union ParameterValue
+{
+    uint8_t i8;
+    uint16_t i16;
+    uint32_t i32;
+    uint64_t i64;
+    float f32;
```
[the remark this answers is not in the document]
Comment (MattTaylorDLS) [reply frameProcessor/include/Frame.h]: I dont think the hdf5 library defines types that you can use. It has the hid_t types like H5T_NATIVE_UINT64, but the thing you pass to the function is just a void* with one of those hid_t vars to say how many bytes it is.

I don't think there's an equivalent to uintxx_t for floats, not that i can find anyway.

[the remark this answers is not in the document]
Comment (MattTaylorDLS) [reply frameProcessor/include/Frame.h]: The hdf5 type doesn't specify what kind of float it is either, it's just H5T_NATIVE_FLOAT. So maybe i should remove all references to it being a 32 float, and just call it 'flt', or 'floating', or something else along those lines

**Comment by MattTaylorDLS** [comment]
Think this is now at a stage where I'm happy to commit

**Comment by MattTaylorDLS** [comment]
...though I still have questions about the UID specific implementation of this. But this PR can stand alone as the implementation of parameter datasets

**Comment by MattTaylorDLS** [comment]
Ah yes, i've moved it in to a separate file. I've also created a unit test for it but it's meant i've had to make the process_frame() method in the UIDAdjustmentPlugin public...

**Pull request merged**